service/src/main/java/bio/terra/profile/service/azure/ApplicationService.java

```diff
@@ -67,4 +68,29 @@ public void addTagToMrg(
     tags.put(tag, value);
     crlService.updateTagsForResource(tenantId, subscriptionId, mrgResource.id(), tags);
   }
+
+  /**
+   * Removes a tag to the MRG associated with a given profile.
+   *
+   * @param tenantId Tenant ID associated with the MRG
+   * @param subscriptionId Subscription ID associated with the MRG
+   * @param managedResourceGroupId ID for the MRG
+   * @param tag Tag name
+   * @throws MissingRequiredFieldsException if the tag is not present.
+   */
+  public void removeTagFromMrg(
+      UUID tenantId, UUID subscriptionId, String managedResourceGroupId, String tag) {
+    var mrgResource = crlService.getResourceGroup(tenantId, subscriptionId, managedResourceGroupId);
+    if (!mrgResource.tags().containsKey(tag)) {
+      throw new MissingRequiredFieldsException(
```

Comment:
Does this exception seem reasonable? Or should I create a more specific one?

Reply:
I'm wondering if we really need to throw an exception in this case. If the tab isn't present in the MRG, then calling this is just a no-op. While perhaps it is unexpected, the end goal is the same (the tag isn't present).

Reply:
That's a good point. Basically framing the operation as "make sure this tag isn't there".

Reply:
+1 to no exception